src/test/java/org/jenkinsci/plugins/durabletask/PowershellScriptTest.java

```diff
@@ -0,0 +1,173 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.durabletask;
+
+import hudson.EnvVars;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.Proc;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.io.IOException;
+import hudson.util.StreamTaskListener;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import org.apache.commons.io.output.TeeOutputStream;
+import static org.hamcrest.Matchers.containsString;
+import org.junit.Test;
+import static org.junit.Assert.*;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import java.util.Properties;
+
+public class PowershellScriptTest {	
+    @Rule public JenkinsRule j = new JenkinsRule();
+
+    private StreamTaskListener listener;
+    private FilePath ws;
+    private Launcher launcher;
+
+    @Before public void vars() throws IOException, InterruptedException {
+        listener = StreamTaskListener.fromStdout();
+        ws = j.jenkins.getRootPath().child("ws");
+        launcher = j.jenkins.createLauncher(listener);
+        Properties properties = System.getProperties();
+        String pathSeparator = properties.getProperty("path.separator");
+        String[] paths = System.getenv("PATH").split(pathSeparator);
+        boolean powershellExists = false;
+        String cmd = launcher.isUnix()?"powershell":"powershell.exe";
+        for (String p : paths) {
+            File f = new File(p, cmd);
+            if (f.exists()) {
+                powershellExists = true;
+                break;
+            }
+        }
+        Assume.assumeTrue("This test should only run if powershell is available", powershellExists == true);
+        
+        // Assume Powershell major version is at least 3
+        if (powershellExists == true) {
+            List<String> args = new ArrayList<String>();
+            args.addAll(Arrays.asList(cmd,"$PSVersionTable.PSVersion.Major"));
```

Comment:
ExecutionPolicy is not a thing in PowerShell for Linux. It's reported as a bug in PowerShells GitHub issue tracker, but it's not going to be fixed because it's not expected to have any impact on Linux. Most likely the argument is ignored, but I think it should be omitted as a sanity measure. https://github.com/PowerShell/PowerShell/issues/478